src/transformers/models/yolos/image_processing_yolos.py

```diff
@@ -678,8 +714,16 @@ class YolosImageProcessor(BaseImageProcessor):
             Controls whether to resize the image's (height, width) dimensions to the specified `size`. Can be
             overridden by the `do_resize` parameter in the `preprocess` method.
         size (`Dict[str, int]` *optional*, defaults to `{"shortest_edge": 800, "longest_edge": 1333}`):
-            Size of the image's (height, width) dimensions after resizing. Can be overridden by the `size` parameter in
-            the `preprocess` method.
+            Size of the image's `(height, width)` dimensions after resizing. Can be overridden by the `size` parameter
+            in the `preprocess` method. Available options are:
+                - `{"height": int, "width": int}`: The image will be resized to the exact size `(height, width)`.
+                    Do NOT keep the aspect ratio.
+                - `{"shortest_edge": int, "longest_edge": int}`: The image will be resized to a maximum size respecting
+                    the aspect ratio and keeping the shortest edge less or equal to `shortest_edge` and the longest edge
+                    less or equal to `longest_edge`.
+                - `{"max_height": int, "max_width": int}`: The image will be resized to the maximum size respecting the
+                    aspect ratio and keeping the height less or equal to `max_height` and the width less or equal to
+                    `max_width`.
```

Comment:
As max_size no longer controls the padding, I think we don't need to include this argument at the moment, but could add in a separate follow-up PR, as the default `shortest_edge` and `longest_edge` already provide bounds. Is it something you found to be useful when finetuning with the object detection script?

Reply:
`{"max_height": int, "max_width": int}` provides a more controllable way of resizing, it is not necessary, but it allows us to use non-square `pad_size`.

For example, we have two images in a batch with shapes
```python
[
    [400, 600],
    [600, 400],
]
```
If we specify `size={"shortest_edge": 200, "longest_edge": 300}` we have to set `pad_size={"height": 300, "width": 300}` (square), because we don't know height or width, will be resized to "largest_edge".
```python
[
    # original -> resized -> padded
    [400, 600] -> [200, 300] -> [300, 300]
    [600, 400] -> [300, 200] -> [300, 300]
]
```
If we specify `size={"max_height": 200, "max_width": 300}` we can set `pad_size={"height": 200, "width": 300}` (non-square).
```python
[
    # original -> resized -> padded
    [400, 600] -> [200, 300] -> [200, 300]
    [600, 400] -> [200, 133] -> [200, 300]
]
```
I don't think it is very common case, but might be useful in cases when you expect most of the images to have a particular orientation.

Reply:
OK, let's keep it!